### ISSUE TITLE
DataDog tracking error

### ISSUE THREAD

**Issue by mohit61** [apollographql/router#1789]
I integrated datadog in the apollo-router, after integration I did the load test(400rps), and I started getting following errors:
`ERROR apollo_router::plugins::telemetry: OpenTelemetry trace error occurred: cannot send span to the batch span processor because the channel is full`

Machine Info
 - OS: CentOS
 - No. of cores: 8
 - Memory: 15G

This is my router config for datadog:
```
telemetry:
  tracing:
    trace_config:
      service_name: router
    datadog:
      endpoint: default
```

Below is the cpu and memory info when error starts occuring:
<img width="627" alt="cpu_memory_router" src="https://user-images.githubusercontent.com/32329331/190356829-78425ef8-10fe-40e8-a708-76d22127a6b8.png">

**Comment by Geal** [comment]
this is due to limitations with the opentelemetry client in the router. The router can handle more traffic than the opentelemetry part can support. You can work around it by sampling the traces, as follows:

```yaml
telemetry:
  tracing:
    trace_config:
      service_name: router
      sampler: 0.01 # 1% of the traces will be sent
    datadog:
      endpoint: default
```

**Comment by BrynCooke** [comment]
In addition you can modify the behaviour of BatchSpanProcessor using the following env variables: https://opentelemetry.io/docs/reference/specification/sdk-environment-variables/#batch-span-processor

In future versions of the router we are looking to expose this config via the router.yaml.

**Comment by BrynCooke** [comment]
Blocked on a new release of Otel

**Comment by BrynCooke** [comment]
This should be fixed now as part of #2232
